File: src/main/java/com/saeed/tasky/models/dto/mapper/ProjectMapper.java
package com.saeed.tasky.models.dto.mapper;

import com.saeed.tasky.models.Project;
import com.saeed.tasky.models.User;
import com.saeed.tasky.models.dto.ProjectDto;
import com.saeed.tasky.models.dto.UserDto;
import org.mapstruct.Mapper;
import org.mapstruct.ReportingPolicy;

@Mapper(componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface ProjectMapper {

    ProjectDto mapProjectToDto(Project project);

    Project mapDtoToProject(ProjectDto projectDto);
}

File: src/main/java/com/saeed/tasky/models/dto/mapper/ProjectMapper.kt
package com.saeed.tasky.models.dto.mapper

import com.saeed.tasky.models.Project
import com.saeed.tasky.models.dto.ProjectDto

object ProjectMapper {

    fun mapProjectToDto(project: Project): ProjectDto {
        val dto = ProjectDto()
        dto.projectName = project.name
        return dto
    }

    fun mapDtoToProject(dto: ProjectDto): Project {
        val project = Project()
        project.name = dto.projectName
        project.icon = dto.projectIcon
        return project
    }
}